redis.go

```diff
@@ -215,42 +225,88 @@ func createLogstashMessage(m *router.Message, docker_host string, use_v0 bool, l
 	name := m.Container.Name[1:]
 	timestamp := m.Time.UTC().Format(time.RFC3339Nano)
 
-	var msg interface{}
-
-	if use_v0 {
-		msg = LogstashMessageV0{
-			Type:       logstash_type,
-			Message:    m.Data,
-			Timestamp:  timestamp,
-			Sourcehost: m.Container.Config.Hostname,
-			Fields: LogstashFields{
-				Docker: DockerFields{
-					CID:        cid,
-					Name:       name,
-					Image:      image,
-					ImageTag:   image_tag,
-					Source:     m.Source,
-					DockerHost: docker_host,
-				},
-			},
+	msg := LogstashMessageV1{}
+
+	msg.Type = logstash_type
+	msg.Timestamp = timestamp
+	msg.Sourcehost = m.Container.Config.Hostname
+	msg.Fields.CID = cid
+	msg.Fields.Name = name
+	msg.Fields.Image = image
+	msg.Fields.ImageTag = image_tag
+	msg.Fields.Source = m.Source
+	msg.Fields.DockerHost = docker_host
+
+	// Check if the message to log itself is json
+	if validJsonMessage(m.Data) {
+		// So it is, include it in the LogstashmessageV1
+		err := msg.UnmarshalDynamicJSON([]byte(m.Data))
+		if err != nil {
+			// Can't unmarshall the json (invalid?), put it in message
+			msg.Message = m.Data
+		} else if msg.Message == "" {
+			msg.Message = NO_MESSAGE_PROVIDED
 		}
 	} else {
-		msg = LogstashMessageV1{
-			Type:       logstash_type,
-			Message:    m.Data,
-			Timestamp:  timestamp,
-			Sourcehost: m.Container.Config.Hostname,
-			Fields: DockerFields{
-				CID:        cid,
-				Name:       name,
-				Image:      image,
-				ImageTag:   image_tag,
-				Source:     m.Source,
-				DockerHost: docker_host,
-			},
-		}
+		// Regular logging (no json)
+		msg.Message = m.Data
 	}
 
 	return json.Marshal(msg)
 
 }
+
+func validJsonMessage(s string) bool {
+
+	if !strings.HasPrefix(s, "{") || !strings.HasSuffix(s, "}") {
```

Comment:
not sure if its needed but maybe right-trim whitespace before checking for the closing curly?